pkg/customresourcestate/custom_resource_metrics_test.go

```diff
@@ -58,18 +130,118 @@ func TestNewCustomResourceMetrics(t *testing.T) {
 						},
 					},
 				},
+				MetricNamePrefix: pointer.String("apps_deployment"),
 			},
 			wantErr: false,
+			wantResult: &customResourceMetrics{
+				MetricNamePrefix: "apps_deployment",
+				GroupVersionKind: schema.GroupVersionKind{
+					Group:   "apps",
+					Version: "v1",
+					Kind:    "Deployment",
+				},
+				ResourceName: "deployments",
+				Families: []compiledFamily{
+					{
+						Name: "apps_deployment_test_metrics",
+						Help: "metrics for testing",
+						Each: &compiledInfo{},
+						Labels: map[string]string{
+							"customresource_group":   "apps",
+							"customresource_kind":    "Deployment",
+							"customresource_version": "v1",
+							"hello":                  "world",
+						},
+						LabelFromPath: map[string]valuePath{
+							"name": mustCompilePath(t, "metadata", "name"),
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "cr metric with custom prefix - expect error",
+			r: Resource{
+				GroupVersionKind: GroupVersionKind{
+					Group:   "apps",
+					Version: "v1",
+					Kind:    "Deployment",
+				},
+				Labels: Labels{
+					LabelsFromPath: map[string][]string{
+						"name": {"metadata", "name"},
+					},
+					CommonLabels: map[string]string{
+						"hello": "world",
+					},
+				},
+				Metrics: []Generator{
+					{
+						Name: "test_metrics",
+						Help: "metrics for testing",
+						Each: Metric{
+							Type: MetricTypeInfo,
+							Info: &MetricInfo{
+								MetricMeta: MetricMeta{
+									Path: []string{
+										"metadata",
+										"annotations",
+									},
+								},
+								LabelFromKey: "test",
+							},
+						},
+					},
+				},
+				MetricNamePrefix: pointer.String("apps_deployment"),
+			},
+			wantErr: true,
+			wantResult: &customResourceMetrics{
+				GroupVersionKind: schema.GroupVersionKind{
+					Group:   "apps",
+					Version: "v1",
+					Kind:    "Deployment",
+				},
+				ResourceName: "deployments",
+				Families: []compiledFamily{
+					{
+						Name: "apps_deployment_test_metrics",
+						Help: "metrics for testing",
+						Each: &compiledInfo{},
+						Labels: map[string]string{
+							"customresource_group":   "apps",
```

Comment:
customresource_group to cr_group?

Reply:
comming from [this comment](https://github.com/kubernetes/kube-state-metrics/pull/1942#discussion_r1061934899)

> I agree, would be nice to have one thing. I believe right now we mix Custom Resource, Custom Resource Definition and Custom Resource State. `kube_customresource` would align with `kube_horizontalpodautoscaler` (no additional underscores).
> 
> Should we rename everything custom resource state into simply custom resource and abbreviate it with "cr" where necessary?

i've started with `customresource` (because i like it a bit more) but as i'm not very opinionated i'm also fine to switch to `cr`

Reply:
cc @mrueg which label name do you prefer? `cr_group` or `customresource_group`

Reply:
My personal preference is the second one, two letter acronyms can become ambiguous